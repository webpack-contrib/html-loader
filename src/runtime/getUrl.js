module.exports = (url, options) => {
  if (!options) {
    // eslint-disable-next-line no-param-reassign
    options = {};
  }

  if (!url) {
    return url;
  }

  // eslint-disable-next-line no-underscore-dangle, no-param-reassign
  url = String(url.__esModule ? url.default : url);

  if (options.hash) {
    // eslint-disable-next-line no-param-reassign
    url += options.hash;
  }

  if (options.maybeNeedQuotes && /[\t\n\f\r "'=<>`]/.test(url)) {
    return `"${url}"`;
  }

  return url;
};
