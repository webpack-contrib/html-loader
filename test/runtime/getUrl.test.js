import getUrl from "../../src/runtime/getUrl";

describe("getUrl", () => {
  it("should work", () => {
    expect(getUrl(true)).toMatchSnapshot();
    expect(getUrl(null)).toMatchSnapshot();
    // eslint-disable-next-line no-undefined
    expect(getUrl(undefined)).toMatchSnapshot();
    expect(getUrl("image.png")).toMatchSnapshot();
    expect(getUrl("image\timage.png")).toMatchSnapshot();
    expect(getUrl("image\nimage.png")).toMatchSnapshot();
    expect(getUrl("image\fimage.png")).toMatchSnapshot();
    expect(getUrl("image\rimage.png")).toMatchSnapshot();
    expect(getUrl("image image.png")).toMatchSnapshot();
    expect(getUrl('image"image.png')).toMatchSnapshot();
    expect(getUrl("image'image.png")).toMatchSnapshot();
    expect(getUrl("image=image.png")).toMatchSnapshot();
    expect(getUrl("image>image.png")).toMatchSnapshot();
    expect(getUrl("image<image.png")).toMatchSnapshot();
    expect(getUrl("image`image.png")).toMatchSnapshot();
    expect(getUrl("image.png", { maybeNeedQuotes: true })).toMatchSnapshot();
    expect(
      getUrl("image\timage.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image\nimage.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image\fimage.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image\rimage.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl('image"image.png', { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image'image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image=image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image>image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image<image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image`image.png", { maybeNeedQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image\timage.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image\nimage.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image\fimage.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image\rimage.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl('image"image.png', { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image'image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image=image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image>image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image<image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl("image`image.png", { maybeNeedQuotes: true, hash: "#hash" })
    ).toMatchSnapshot();
    expect(getUrl("image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image\timage.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image\nimage.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image\fimage.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image\rimage.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl('image"image.png', { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image'image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image=image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image>image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image<image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image`image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(
      getUrl({ default: "image.png", __esModule: true })
    ).toMatchSnapshot();
  });
});
