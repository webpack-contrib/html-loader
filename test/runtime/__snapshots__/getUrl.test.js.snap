// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUrl should work 1`] = `"true"`;

exports[`getUrl should work 2`] = `null`;

exports[`getUrl should work 3`] = `undefined`;

exports[`getUrl should work 4`] = `"image.png"`;

exports[`getUrl should work 5`] = `"image	image.png"`;

exports[`getUrl should work 6`] = `
"image
image.png"
`;

exports[`getUrl should work 7`] = `"imageimage.png"`;

exports[`getUrl should work 8`] = `
"image
image.png"
`;

exports[`getUrl should work 9`] = `"image image.png"`;

exports[`getUrl should work 10`] = `"image"image.png"`;

exports[`getUrl should work 11`] = `"image'image.png"`;

exports[`getUrl should work 12`] = `"image=image.png"`;

exports[`getUrl should work 13`] = `"image>image.png"`;

exports[`getUrl should work 14`] = `"image<image.png"`;

exports[`getUrl should work 15`] = `"image\`image.png"`;

exports[`getUrl should work 16`] = `"image.png"`;

exports[`getUrl should work 17`] = `""image	image.png""`;

exports[`getUrl should work 18`] = `
""image
image.png""
`;

exports[`getUrl should work 19`] = `""imageimage.png""`;

exports[`getUrl should work 20`] = `
""image
image.png""
`;

exports[`getUrl should work 21`] = `""image image.png""`;

exports[`getUrl should work 22`] = `""image"image.png""`;

exports[`getUrl should work 23`] = `""image'image.png""`;

exports[`getUrl should work 24`] = `""image=image.png""`;

exports[`getUrl should work 25`] = `""image>image.png""`;

exports[`getUrl should work 26`] = `""image<image.png""`;

exports[`getUrl should work 27`] = `""image\`image.png""`;

exports[`getUrl should work 28`] = `"image.png#hash"`;

exports[`getUrl should work 29`] = `""image	image.png#hash""`;

exports[`getUrl should work 30`] = `
""image
image.png#hash""
`;

exports[`getUrl should work 31`] = `""imageimage.png#hash""`;

exports[`getUrl should work 32`] = `
""image
image.png#hash""
`;

exports[`getUrl should work 33`] = `""image image.png#hash""`;

exports[`getUrl should work 34`] = `""image"image.png#hash""`;

exports[`getUrl should work 35`] = `""image'image.png#hash""`;

exports[`getUrl should work 36`] = `""image=image.png#hash""`;

exports[`getUrl should work 37`] = `""image>image.png#hash""`;

exports[`getUrl should work 38`] = `""image<image.png#hash""`;

exports[`getUrl should work 39`] = `""image\`image.png#hash""`;

exports[`getUrl should work 40`] = `"image.png#hash"`;

exports[`getUrl should work 41`] = `"image	image.png#hash"`;

exports[`getUrl should work 42`] = `
"image
image.png#hash"
`;

exports[`getUrl should work 43`] = `"imageimage.png#hash"`;

exports[`getUrl should work 44`] = `
"image
image.png#hash"
`;

exports[`getUrl should work 45`] = `"image image.png#hash"`;

exports[`getUrl should work 46`] = `"image"image.png#hash"`;

exports[`getUrl should work 47`] = `"image'image.png#hash"`;

exports[`getUrl should work 48`] = `"image=image.png#hash"`;

exports[`getUrl should work 49`] = `"image>image.png#hash"`;

exports[`getUrl should work 50`] = `"image<image.png#hash"`;

exports[`getUrl should work 51`] = `"image\`image.png#hash"`;

exports[`getUrl should work 52`] = `"image.png"`;
